xarray/core/variable.py

```diff
@@ -993,6 +993,32 @@ def chunk(self, chunks=None, name=None, lock=False):
 
         return type(self)(self.dims, data, self._attrs, self._encoding, fastpath=True)
 
+    def _as_sparse(self, sparse_format=_default, fill_value=dtypes.NA):
+        """
+        use sparse-array as backend.
+        """
+        import sparse
+
+        # TODO  what to do if dask-backended?
+        if fill_value is dtypes.NA:
+            dtype, fill_value = dtypes.maybe_promote(self.dtype)
+        else:
+            dtype = self.dtype
+
+        if sparse_format is _default:
+            sparse_format = "coo"
+        as_sparse = getattr(sparse, "as_{}".format(sparse_format.lower()))
+        data = as_sparse(self.data.astype(dtype), fill_value=fill_value)
+        return self._replace(data=data)
+
+    def _to_dense(self):
```

Comment:
Also private, as is `_as_sparse`.

Reply:
We should make these public in `DataArray` and `Dataset`. See discussion here: https://github.com/pydata/xarray/issues/3245#issuecomment-524407909

Can be left for a future PR though :)

Reply:
Thanks, @dcherian. 
I would like to expose them to public, but what is the best name of these functions? 
#3245